test/e2e/helpers/broker_with_config_helper.go

```diff
@@ -67,36 +69,50 @@ func TestBrokerWithConfig(t *testing.T,
 		client.CreateBrokerV1Beta1OrFail(brokerName, resources.WithBrokerClassForBrokerV1Beta1(brokerClass), resources.WithConfigForBrokerV1Beta1(config))
 		client.WaitForResourceReadyOrFail(brokerName, lib.BrokerTypeMeta)
 
-		// create the event we want to transform to
-		transformedEventBody := fmt.Sprintf("%s %s", eventBody, string(uuid.NewUUID()))
-		eventAfterTransformation := cloudevents.New(
-			fmt.Sprintf(`{"msg":%q}`, transformedEventBody),
-			cloudevents.WithSource(eventSource2),
-			cloudevents.WithType(eventType2),
-		)
+		// eventToSend is the event sent as input of the test
+		eventToSend := cloudevents.NewEvent()
+		eventToSend.SetID(uuid.New().String())
+		eventToSend.SetType(eventType)
+		eventToSend.SetSource(eventSource)
+		if err := eventToSend.SetData(cloudevents.ApplicationJSON, []byte(eventBody)); err != nil {
+			t.Fatalf("Cannot set the payload of the event: %s", err.Error())
+		}
+
+		transformedEvent := cloudevents.NewEvent()
```

Comment:
It does not seem necessary to create the transformedEvetn here with the changes to resources.EventTransformationPod

Reply:
That's true!